starknet_py/net/schemas/common.py

```diff
@@ -30,8 +30,15 @@ class Felt(fields.Field):
     Field that serializes int to felt (hex encoded string)
     """
 
+    REGEX_PATTERN = r"^0x(0|[a-fA-F1-9]{1}[a-fA-F0-9]{0,62})"
+
     def _serialize(self, value: Any, attr: str, obj: Any, **kwargs):
-        return hex(value)
+        if isinstance(value, int):
+            serialized = hex(value)
+            if re.fullmatch(self.REGEX_PATTERN, serialized) is not None:
+                return serialized
```

Comment:
The primary intent behind this was to avoid double conversion to hex and perform this conversion only when we know that the value is `int`. We can indeed implement the same logic as follows:
```
if isinstance(value, int) and re.fullmatch(self.REGEX_PATTERN, hex(value)):
    return hex(value)
```
Nevertheless, in such cases I personally prefer to explicitly check `is not None`. This is because omitting this condition might suggest that `fullmatch` returns `bool`, which isn't the case.

Reply:
Sure, makes sense, though this is not a part of any external interface, right? So I wouldn't be too bothered about this check, and in this context it does not really matter what fullmatch returns - as long as we can easily verify if there was a hit or not. I get the double conversion bit, this is not ideal.

This is a minor thing, I leave it up to you to decide.